### PR TITLE
Dark mode feature

### PR DESCRIPTION
**Pull request by itpun**
Dark mode is now a feature for HoverZoom. 

Option to set Dark mode is currently in "Advanced" settings. When enabled, the image border color will be set to black and the caption color will be set to white. I can further stylize it later if needed.

I apologize I think i screwd up my branching, this branch might also include my imgur null fix. Please let me know what I can do to fix this if you want!

Ian

### CODE REVIEW

**Comment by JamieSharpe** [comment]
Is it feasible to allow a custom colour to be set?

Then it would also bring in custom themes in general, and not just the hard coded colours.

See issue/request https://github.com/extesy/hoverzoom/issues/72.

**Comment by itpun** [comment]
Hey it's totally possible. Currently I'm wondering if its best to have preset standard of some themes first, or let the user pick whatever colors they want.

**Comment by JamieSharpe** [comment]
Hi,

I think that would be the best approach. Having a small selection of preset colours along with textbox/range sliders for RGB custom colours too.

File: js/hoverzoom.js [hunk 1]
```diff
@@ -449,13 +452,41 @@ var hoverZoom = {
                     video.src = imgDetails.url;
                     imgFullSize = $(video).appendTo(hz.hzImg);
                     video.addEventListener('error', imgFullSizeOnError);
-                    video.addEventListener('loadedmetadata', posImg);
+                    video.addEventListener('loadedmetadata', function(){
+                        posImg();
+                        track = this.addTextTrack("captions", "English", "en");
```
Comment (extesy) [inline js/hoverzoom.js]: Extract this into a function and put behind a new option (disabled by default). Changes like this shouldn't suddenly appear for 280k current users.

File: js/hoverzoom.js [hunk 2]
```diff
@@ -449,13 +452,41 @@ var hoverZoom = {
...
+                                track.addCue(new VTTCue(time, time+1, finalTime)); 
+                                track.cues[time].align ="end";
+                                track.cues[time].position = 100;
+                                track.cues[time].line = 0;
+                            }
+                        }
+                    });
                     video.addEventListener('loadeddata', function() {
-                        imgFullSizeOnLoad();
+                        ();
```
Comment (extesy) [inline js/hoverzoom.js]: what does this mean?

File: js/hoverzoom.js [hunk 3]
```diff
@@ -140,6 +140,9 @@ var hoverZoom = {
                     if (hzCaption.height() > 20) {
                         hzCaption.css('font-weight', 'normal');
                     }
+                    if(options.enableDarkMode) {
```
Comment (extesy) [inline js/hoverzoom.js]: please respect existing code style - have a single space on both sides of parentheses.

File: js/hoverzoom.js [hunk 4]
```diff
@@ -89,7 +89,7 @@ var hoverZoom = {
                 'font':'menu',
                 'font-size':'11px',
                 'font-weight':'bold',
-                'color':'#333',
+                'color':'#000000',
```
Comment (extesy) [inline js/hoverzoom.js]: why change this? pure black color makes contrast too high.

File: plugins/imgur_a.js [hunk 5]
```diff
@@ -76,7 +84,7 @@ hoverZoomPlugins.push({
                                                 caption += ';\n';
                                             }
                                             caption += img.description;
```
Comment (extesy) [inline plugins/imgur_a.js]: I think that "null"s are appended on this line so it's better to check here rather than do a replace later.

**Comment by extesy** [comment]
I've cherry-picked a code to show video timestamps, but the dark feature should be reworked into a more generic color picker, so I'm closing this PR.

**Pull request closed without merging**